### PR TITLE
Added Support for FeedMe Client

### PR DESCRIPTION
**Pull request by marcohald**

### CODE REVIEW

**Review by javerous — APPROVED** [review]

File: xExtension-GReaderRedate/extension.php [hunk 1]
```diff
@@ -39,6 +40,10 @@ public function handle_entry($entry)
 		if (!$redate && $this->isReederEnabled() && (preg_match('#Reeder/[0-9]+#', $user_agent, $matches) >= 1))
 			$redate = true;
 
+		// > Match FeedMe.
+		if (!$redate && $this->isReederEnabled() && (preg_match('#FeedMe/[0-9]+#', $user_agent, $matches) >= 1))
```
Comment (vonguyenkhang) [inline xExtension-GReaderRedate/extension.php:44]: there's a typo here: it should be `isFeedMeEnabled()`

File: xExtension-GReaderRedate/configure.phtml [hunk 2]
```diff
@@ -20,6 +20,13 @@
 			</label>
 		</div>
 
+		<div class="group-controls">
+			<label class="checkbox" for="gr_feedme">
+				<input type="checkbox" id="gr_feedme" name="gr_feedme" value="1" <?= $this->isFeedMeEnabled() ? 'checked' : '' ?>>
+				<?= _t('ext.greader_redate.feed_me') ?>
```
Comment (vonguyenkhang) [inline xExtension-GReaderRedate/configure.phtml:26]: there's a typo here: it should be `<?= _t('ext.greader_redate.feedme') ?>`

File: xExtension-GReaderRedate/i18n/en/ext.php [hunk 3]
```diff
@@ -6,6 +6,7 @@
 		'instruction' => 'Please select at which RSS clients this replacement should be done:',
 		'reeder' => 'Reeder',
 		'easy_rss' => 'EasyRSS',
+		'feedme' => 'FeedMe',
```
Comment (vonguyenkhang) [inline xExtension-GReaderRedate/i18n/en/ext.php:9]: for https://github.com/javerous/freshrss-greader-redate/blob/master/xExtension-GReaderRedate/i18n/fr/ext.php, there's a missing comma after EasyRSS:
```
-        'easy_rss' => 'EasyRSS'
+        'easy_rss' => 'EasyRSS',
+        'feedme' => 'FeedMe',
```

**Comment by vonguyenkhang** [comment]
This implementation doesn't work for FeedMe: FeedMe uses the entry id as the timestamp, so modifying the dateAdded doesn't change the time displayed in FeedMe (somehow FeedMe can't even fetch new entries).

Another workaround is to change the entry id, but this may not be a good idea, because many things will break if the modified entry ids are not unique. 

I think the best way is to convince FeedMe developer to properly use the dateAdded instead of the id for the article's published time.

Related info: https://github.com/FreshRSS/FreshRSS/issues/2759

**Comment by javerous** [comment]
@vonguyenkhang Thank you for your comments. Can you open a new PR as this one have been merged ?

**Comment by vonguyenkhang** [comment]
> @vonguyenkhang Thank you for your comments. Can you open a new PR as this one have been merged ?

After making the changes, somehow FeedMe app is unable to retrieve any entry. I still don't know the reason yet. 
I'll find some time to experiment, but just to let you know I'm no PHP developer and I'm not familiar with FreshRSS internal working or GReader API either, so please don't have high hope. Maybe it's faster if you poke the developer of FeedMe directly.

Tested with FeedMe 3.20.14 on Android, FreshRSS 1.20.0

**Comment by vonguyenkhang** [comment]
> > @vonguyenkhang Thank you for your comments. Can you open a new PR as this one have been merged ?
> 
> After making the changes, somehow FeedMe app is unable to retrieve any entry. I still don't know the reason yet. I'll find some time to experiment, but just to let you know I'm no PHP developer and I'm not familiar with FreshRSS internal working or GReader API either, so please don't have high hope. Maybe it's faster if you poke the developer of FeedMe directly.
> 
> Tested with FeedMe 3.20.14 on Android, FreshRSS 1.20.0

update: I'm still unable to make FeedMe retrieve the entries. Here are what I have tried:
- `$entry->_dateAdded($entry->date(true));` : no item
- `$entry->_dateAdded(1667402880740270,true);` : no item
- `$entry->_dateAdded($entry->date(true),true);` : dates of all entries are 1/1/1970
- `$entry->_dateAdded(1000*($entry->date(true)),true);` : dates of all entries are 1/1/1970
- `$entry->_dateAdded(1000000*($entry->date(true)),true);` : no item

FeedMe is not open-source, so it's not straightforward to find out what's the cause of the problem.

My FreshRSS instance is running on a DD-WRT router, so there's also a chance that the problem is platform-specific.

I'm unable to make any real progress on this, so I'll check with FeedMe developer. I see this is an open issue at FeedMe repo but it's not been updated for 1 and a half years. https://github.com/seazon/FeedMe/issues/328

**Pull request merged**